src/js/actions/mask.js

```diff
@@ -142,6 +150,98 @@ define(function (require, exports) {
             menuActions.native]
     };
 
+    /**
+     * Create an circular mask the size of the currently selected layer
+     *
+     * @return {Promise}
+     */
+    var applyRectangle = function () {
+        var appStore = this.flux.store("application"),
+            toolStore = this.flux.store("tool"),
+            currentDocument = appStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        var currentLayers = currentDocument.layers.selected;
+
+        if (currentLayers.isEmpty()) {
+            return Promise.resolve();
+        }
+
+        var currentLayer = currentLayers.first(),
+            bounds = boundsLib.bounds(currentLayer.bounds),
+            options = {
+                historyStateInfo: {
+                    name: nls.localize("strings.ACTIONS.ADD_VECTOR_MASK"),
+                    target: documentLib.referenceBy.id(currentDocument.id),
+                    coalesce: true,
+                    suppressHistoryStateNotification: true
+                }
+            };
+
+        return descriptor.batchPlayObjects([vectorMaskLib.makeBoundsWorkPath(bounds),
+            vectorMaskLib.makeVectorMaskFromWorkPath(),
```

Comment:
Can you indent these two lines so they're obviously part of the call one line higher, and not in the same indent with the `.bind` and `.then`. The code flow here makes it hard to read.